File: weather_script.py
import requests

def get_weather(lat, lon, api_key):
    url = f"https://api.openweathermap.org/data/2.5/weather?lat={lat}&lon={lon}&appid={api_key}"
    response = requests.get(url)
    
    if response.status_code == 200:
        data = response.json()
        main = data['weather'][0]['main']
        description = data['weather'][0]['description']
        temp = data['main']['temp']
        print(f"Weather: {main}")
        print(f"Description: {description}")
        print(f"Temperature: {temp}K")
    else:
        print("Error in the HTTP request")

if __name__ == "__main__":
    lat = input("Enter latitude: ")
    lon = input("Enter longitude: ")
    api_key = input("Enter your API key: ")
    get_weather(lat, lon, api_key)

File: weather_script.js
const apiKey = "5a92764b0d773276747a0050d248fb61"; // Replace with your OpenWeatherMap API key
const apiUrl = "https://api.openweathermap.org/data/2.5/weather";

async function fetchWeather(city) {
    try {
        const response = await fetch(`${apiUrl}?q=${city}&appid=${apiKey}&units=metric`);
        if (!response.ok) {
            throw new Error("City not found or API error.");
        }
        const data = await response.json();
        displayWeather(data);
    } catch (error) {
        console.error("Error fetching weather data:", error.message);
        alert("Error fetching weather data. Please try again.");
    }
}

function displayWeather(data) {
    const { name, main, weather } = data;
    const temperature = main.temp;
    const description = weather[0].description;

    alert(`Weather in ${name}:
    Temperature: ${temperature}°C
    Description: ${description}`);
}

// Example usage
const city = prompt("Enter a city name:");
if (city) {
    fetchWeather(city);
}